mozdownload/scraper.py

```diff
@@ -696,7 +696,14 @@ def get_build_info(self):
         self.show_matching_builds(parser.entries)
         self.builds = parser.entries
         self.build_index = len(parser.entries) - 1
-
+        
+        if self.build_number:
+          self.tmp = "build"+str(self.build_number)
+          if self.tmp not in self.builds:
+            message = 'Selected build is not available.'
+            sys.logger.info(message)
+            sys.exit()
```

Comment:
Never call `sys.exit()` but raise an exception. Have a look at the ones mozdownload already defines.